### PR TITLE
update azure-sdk-for-go to v56.0.0

### PR DESCRIPTION
**Pull request by ms-henglu**
Tc tests. Though there's a failed test, but it's irrelevant.

image

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ cloud.google.com/go/internal/version
 # cloud.google.com/go/storage v1.16.0
 ## explicit
 cloud.google.com/go/storage
-# github.com/Azure/azure-sdk-for-go v55.6.0+incompatible
+# github.com/Azure/azure-sdk-for-go v56.0.0+incompatible
```
Comment (tombuildsstuff) [inline vendor/modules.txt:12]: this major SDK upgrade needs to be done in a separate PR and then this upgraded - can we pull this out?

Comment (ms-henglu) [reply vendor/modules.txt:12]: Hi Tom, thanks, I'll only update azure-sdk-for-go in this PR.

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍
